### ISSUE TITLE
Cannot switch Static DisposableVM to new DisposableVM Template

### ISSUE THREAD

**Issue by andrewdavidwong** [QubesOS/qubes-issues#5491]
**Qubes OS version**

`R4.0`

**Affected component(s) or functionality**

Core

**Brief summary**

There is no way to switch a Static DisposableVM to a new DisposableVM Template, or the system does not give any hint for how to do it.

**To Reproduce**

Suppose you have a Static DisposableVM named `printer` based on a DisposableVM Template named `fedora-29-dvm` based on a TemplateVM named `fedora-29`. Now, Fedora 29 has reached EOL, so you want to upgrade to Fedora 30.

1. Clone `fedora-29` template to `fedora-30`.
2. Upgrade `fedora-30` template to Fedora 30.
3. Attempt to switch `fedora-29-dvm` from `fedora-29` to `fedora-30`.
4. Receive this message:
```
ERROR: Basic tab:
Cannot change template while there are DispVMs based on this qube
```
5. Attempt to switch `printer` to a different DisposableVM Template. There is no way to do this. The option is grayed-out in Qube Manager, and the command line says:
```
qvm-prefs -s printer template fedora-30
usage: qvm-prefs [--verbose] [--quiet] [--help] [--help-properties]
                 [--hide-default] [--get] [--set] [--default]
                 VMNAME [PROPERTY] [VALUE]
qvm-prefs: error: no such property: 'template'
```

**Expected behavior**

There is some possible way to switch a Static DisposableVM to a new DisposableVM Template.

**Actual behavior**

There is no possible way, or else the system does not offer any hint for how to do it.

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**

https://www.qubes-os.org/doc/disposablevm-customization

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**

None found.

**Comment by hexagonrecursion** [comment]
related: https://github.com/QubesOS/qubes-issues/issues/3576#issuecomment-365221714 https://github.com/QubesOS/qubes-core-admin/commit/717bc4ace3e1830bf283861368352cc8545f3275

**Comment by zpc0** [comment]
Since this problem also affect R4.0, could you back-port related fixes to R4.0 package?

**Comment by andrewdavidwong** [comment]
> Since this problem also affect R4.0, could you back-port related fixes to R4.0 package?

This issue is already on the "Release 4.0 updates" milestone.

**Comment by tlaurion** [comment]
Normal that this issue is closed?

**Comment by tlaurion** [comment]
https://github.com/QubesOS/qubes-issues/issues/3576#issuecomment-457051729

**Comment by andrewdavidwong** [comment]
> Normal that this issue is closed?

Sorry, I don't understand what you mean. Here's the message above stating how it was closed:
> @marmarek closed this in QubesOS/qubes-core-admin#374 11 days ago

Does that help?

**Comment by marmarek** [comment]
I think @tlaurion asked why this issue is closed while the issue is not fixed yet in a version mentioned in "milestone" field.
Answer: yes, this is normal. Github doesn't allow per-version issue state, so the issue state tracks "master" branch - is closed as soon as fix is merged there. But then we backport the fix to older versions where applicable. This is the workflow that allows us to use supported automation (linking fixes to issues etc) to the greatest possible extend we can think of.
